src/main/java/io/cdap/plugin/gcp/publisher/source/PubSubSubscriberUtil.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcp.publisher.source;
+
+import com.google.auth.Credentials;
+import io.cdap.cdap.etl.api.streaming.StreamingContext;
+import io.cdap.plugin.gcp.common.GCPUtils;
+import org.apache.spark.storage.StorageLevel;
+import org.apache.spark.streaming.api.java.JavaDStream;
+import org.apache.spark.streaming.api.java.JavaReceiverInputDStream;
+import org.apache.spark.streaming.dstream.ReceiverInputDStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Serializable;
+import scala.reflect.ClassTag;
+
+import java.util.ArrayList;
+
+/**
+ * Utility class to create a JavaDStream of received messages.
+ */
+public class PubSubSubscriberUtil implements Serializable {
```

Comment:
Why a util class needs to be serializable?

Reply:
Oops, good catch.